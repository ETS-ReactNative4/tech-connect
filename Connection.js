import React, { Component } from 'react'
import { View, Text, Image, StyleSheet, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'
import { connect } from 'react-redux'

export class Connection extends Component {
  
  render() {

    const { connection } = this.props

    return (
      <View style={ styles.container }>
        <Image source={ require('./profile-pic.jpeg')} style={styles.picture} />
        <View style={ styles.connectionContainer }>
          <TouchableOpacity onPress={() => this.props.viewProfile(connection.id)}>
            <Text style={styles.name}>{ connection.name }</Text>
          </TouchableOpacity>
          <Text style={ styles.position }>{ connection.job_title }</Text>
          <View style={ styles.locationContainer}>
            <Icon name='map-pin' size={18} color='#4AA9C5' style={{marginRight: 7}} />
            <Text style={ styles.location }>{ connection.city }</Text>
          </View>
        </View>
      </View>
    ) 
  }
}

export const mapStateToProps = (state) => ({
  api_key: state.user.api_key
})

export default connect(mapStateToProps)(Connection)

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    display: 'flex',
    flexDirection: 'row',
    padding: 10,
    borderRadius: 5,
    marginBottom: 10,
    shadowOffset: {  width: 2,  height: 2},
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  picture: {
    height: 80,
    width: 80,
    borderRadius: 40
  },
  name: {
    fontSize: 18
  },
  position: {
    color: 'gray',
    marginTop: 1
  },
  location: {
    color: 'gray',
    fontSize: 14,
    marginTop: 2
  },
  connectionContainer: {
    marginLeft: 5
  },
  locationContainer: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: 3
  }
})