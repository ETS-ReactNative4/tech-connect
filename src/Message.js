import React, { Component } from 'react'
import { StyleSheet, View, Image, Text, TouchableOpacity } from 'react-native'
import { connect } from 'react-redux';


export class Message extends Component {
  constructor() {
    super()
    this.state = {
      showMessage: false
    }
  }

  render() {
    const { message, user } = this.props
    const connection = (user.name === message.receiver.name) ? message.sender.name :  message.receiver.name
    const sentDate = new Date(message.created_at).toUTCString().split(' ').slice(0, 4).join(' ')
    const meetingDate = new Date(message.meeting_date).toUTCString().split(' ').slice(0, 4).join(' ')
    const time = new Date(message.meeting_date).toUTCString().split(' ')[4]
    let messageText
    let messagePhoto
    let messageEnd = (message.meeting_location !== 'N/A') ? `at ${message.meeting_location} on ${meetingDate} at ${time}` : null

    if (message.receiver.name === user.name) {
      messageText = <Text style={ styles.messageBody } >You have { message.status } a meeting with { message.sender.name } { messageEnd }</Text>
      messagePhoto = message.sender.photo
    } else {
      messageText = <Text style={ styles.messageBody } >{ message.receiver.name } has { message.status } a meeting with you { messageEnd }</Text>
      messagePhoto = message.receiver.photo
    }
    return (
      <View style={ styles.messageContainer }>
        <Text style={ styles.date }>{ sentDate }</Text>
        <TouchableOpacity style={ styles.messageInfoContainer } onPress={() => this.setState({showMessage: !this.state.showMessage})}>
          <View style={ styles.imageContainer }>
            <Image source={ {uri: messagePhoto} } style={ styles.userPic }/>
          </View>
          <View style={ styles.messageInfo }>
            <Text style={ styles.name }>{ connection }</Text>
            <Text style={ styles.message }>{ message.status }</Text>
            {
              !this.state.showMessage ? null : messageText
            }
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(Message)


const styles = StyleSheet.create({
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    padding: 20,
    borderBottomWidth: 0.5,
    borderBottomColor: 'gray',
  },
  messageInfoContainer: {
    display: 'flex',
    flexDirection: 'row',
    maxWidth: '100%',
    marginRight: 10
  },
  imageContainer: {
    shadowOffset: {  width: 2,  height: 2 },
    shadowRadius: 4,
    shadowColor: 'black',
    shadowOpacity: .2,
  },
  userPic: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  date: {
    position: 'absolute',
    right: 20,
    top: 20
  },
  name: {
    fontSize: 18
  },
  message: {
    marginTop: 5,
    color: 'gray',
  },
  messageBody: {
    marginRight: 10
  },
  messageInfo: {
    marginLeft: 10,
    marginRight: 30,
    marginTop: 5
  }
})