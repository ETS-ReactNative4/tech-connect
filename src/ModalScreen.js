import React, { Component } from 'react'
import { StyleSheet, Text, View, TextInput, Button, TouchableHighlight, TouchableOpacity } from 'react-native';
import { Input } from 'react-native-elements'
import DatePicker from 'react-native-datepicker'
import { connect } from 'react-redux'
import { sendConnectionRequest } from '../apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class ModalScreen extends Component {
  constructor() {
    super()
    this.state = {
      response: '',
      inputs: {
        datetime_1: '', 
        datetime_2: '', 
        datetime_3: '',
        meeting_1: '',
        meeting_2: '',
        meeting_3: ''   
      }
    }
  }

  sendRequest = async () => {
    const user = this.props.navigation.getParam('user')
    const response = await sendConnectionRequest(this.props.user.api_key, user.id, this.state.inputs)
    this.setState({ response })
  }

  render() {
    const { response, inputs } = this.state
    const { datetime_1, datetime_2, datetime_3, meeting_1, meeting_2, meeting_3 } = this.state.inputs
    const user = this.props.navigation.getParam('user')
    const note = `${user.name} will receive an email with your suggestions. Once confirmed you will receive a notification email.`
    const error = response.length ? <Text style={{color: 'red'}}>{response}</Text> : null
    if (response === 'Email Sent') {
      return (
        <View style={styles.container}>
          <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
            <Text style={styles.title}>{ response }</Text>
            <Icon
              name='check-circle'
              size={35}
              color='#fff'
              style={{marginLeft: 5}}
            />
          </View>
          <TouchableHighlight style={styles.btnContainer}>
            <Button
              onPress={() => this.props.navigation.goBack()}
              color={'#fff'}
              title="Done"
            />
          </TouchableHighlight>
        </View>
      )
    } else {
      return (
        <View style={styles.container}>
          <Text style={styles.title}>Request a Meeting</Text>
          <Text style={styles.note}>{ note }</Text>
          <Text style={styles.suggest}>Suggest Three Date/Times:</Text>
          <DatePicker
            style={styles.date}
            date={datetime_1}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_1: date}})}
          />
          <DatePicker
            style={styles.date}
            date={datetime_2}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_2: date}})}
          />
          <DatePicker
            style={styles.date}
            date={datetime_3}
            mode='datetime'
            format='YYYY-MM-DD HH:MM:SS'
            minDate={new Date()}
            maxDate='2020-01-01'
            confirmBtnText='Confirm'
            cancelBtnText='Cancel'
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36,
                borderWidth: 0
              }
            }}
            onDateChange={date => this.setState({inputs: {...this.state.inputs, datetime_3: date}})}
          />
          <Text style={styles.suggest}>Suggest Three Locations:</Text>
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_1: text}})}
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_2: text}})} 
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          <Input 
            onChangeText={text => this.setState({inputs: {...this.state.inputs, meeting_3: text}})}
            containerStyle={styles.input} 
            inputContainerStyle={styles.inputContainer} 
          />
          { error }
          <TouchableHighlight style={styles.btnContainer}>
            <Button 
              onPress={this.sendRequest}
              title="Send Request"
              color={'#fff'}
            />
          </TouchableHighlight>
          <TouchableHighlight style={styles.btnContainer}>
            <Button
              onPress={() => this.props.navigation.goBack()}
              color={'#fff'}
              title="Cancel"
            />
          </TouchableHighlight>
        </View>
      )
    }
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(ModalScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#4AA9C5',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 15
  },
  title: {
    fontSize: 42,
    fontWeight: 'bold',
    color: '#fff',
  },
  note: {
    marginBottom: 25,
    fontSize: 14,
    color: '#fff',
    textAlign: 'center'
  },
  suggest: {
    fontWeight: 'bold',
    fontSize: 22,
    marginTop: 20,
    color: '#fff',
    alignSelf: 'flex-start'
  },
  date: {
    width: '95%', 
    backgroundColor: '#fff', 
    borderRadius: 5, 
    marginTop: 5, 
    marginBottom: 5, 
    borderBottomWidth: 0
  },
  input: {
    width: '95%',
    borderRadius: 5, 
    backgroundColor: '#fff', 
    marginTop: 5,
    marginBottom: 5,
  },
  inputContainer: {
    borderBottomWidth: 0
  }, 
  btnContainer: {
    backgroundColor: '#93548F',
    paddingRight: 10,
    paddingLeft: 10,
    borderRadius: 20,
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 5,
    shadowColor: 'black',
    shadowOpacity: .5,
    marginTop: 10,
    width: 200
  }
})

