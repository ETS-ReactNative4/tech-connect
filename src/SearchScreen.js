import React from 'react';
import { StyleSheet, Text, TextInput, View, Button, TouchableHighlight, Image, ScrollView, ActivityIndicator } from 'react-native';
import { connect } from 'react-redux';
import { ButtonGroup } from 'react-native-elements'
import { LinearGradient } from 'expo'
import Connection from './Connection'
import { getAllUsers, getUserInfo, getUsersFilter } from '../apiCalls'
import Icon from 'react-native-vector-icons/Feather'


export class SearchScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      allUsers: [],
      search: '',
      selectedIndex: 0,
      loading: true
    }
  }

  async componentDidMount() {
    const allUsers = await getAllUsers(this.props.user.api_key)
    this.setState({ allUsers, loading: false })
  }

  viewProfile = async (id) => {
    const user = await getUserInfo(id, this.props.user.api_key)
    this.props.navigation.navigate('ProfilePage', {user})
  }

  updateIndex = (selectedIndex) => {
    this.setState({selectedIndex})
  }

  getFilteredUsers = async () => {
    this.setState({ allUsers: [], loading: true })
    const { selectedIndex, search } = this.state
    const queryParams = ['name', 'city', 'position', 'employer']
    const allUsers = await getUsersFilter(this.props.user.api_key, queryParams[selectedIndex], search)
    this.setState({ allUsers, loading: false })
  }

  render() {
    const { allUsers, selectedIndex, search, loading } = this.state
    const userArray = allUsers.map(user => <Connection connection={ user } viewProfile={this.viewProfile.bind(null,user.id)} key={ user.id }/>)
    const buttons = ['Name', 'Location', 'Position', 'Employer']

    return (
      <View style={styles.container}>
        <View style={styles.componentContainer}>
          <Text style={styles.search}>Search by:</Text>
          <ButtonGroup
            onPress={this.updateIndex}
            selectedIndex={selectedIndex}
            selectedButtonStyle={{backgroundColor: '#4AA9C5'}}
            buttons={buttons}
            containerStyle={{height: 30}}
          />
          <View style={styles.searchContainer}>
            <LinearGradient
              colors={['#4AA9C5', '#6364c1', '#93548F']}
              start={0.4}
              style={styles.gradient} >
              <Icon
                name='search'
                size={16}
                color='#FFF'
                onPress={ this.getFilteredUsers }
              />
            </LinearGradient>
            <TextInput
              style={styles.inputContainer}
              placeholder="Search here..."
              underlineColorAndroid="transparent"
              onChangeText={(text) => this.setState({search: text})} 
              value={search}
            />
          </View>
          <ScrollView contentContainerStyle={{alignItems: 'center'}} style={ styles.scrollContainer }>
            <View style={ styles.connectionsContainer }>
              { !loading ? userArray : <ActivityIndicator size="large" color="#fff" /> }
              { 
                (!userArray.length && !loading) && 
                  <Text style={{color: '#fff'}}>No users were found.</Text> 
              }
            </View>
          </ScrollView>
        </View>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(SearchScreen)


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    paddingTop: 20,
    backgroundColor: '#4AA9C5'
  },
  search: {
    marginTop: 10, 
    marginLeft: 10,
    fontSize: 20
  },
  gradient: {
    marginLeft: 4,
    borderRadius: 300, 
    width: 37, 
    height: 37, 
    justifyContent: 'center', 
    alignItems: 'center'
  },
  componentContainer: {
    flex: 1,
    backgroundColor: '#fff', 
    borderRadius: 5,
    justifyContent: 'center', 
    padding: 15,
    alignItems: 'stretch',
    marginTop: 25,
    marginRight: 20,
    marginLeft: 20,
  },
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10, 
    borderRadius: 30, 
    borderWidth: 2, 
    borderColor: '#4AA9C5',
    backgroundColor: '#ffffff00', 
    borderBottomWidth: 2, 
  },
  inputContainer: {
    flex: 1,
    fontSize: 16,
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 10,
    backgroundColor: 'transparent'
  },
  scrollContainer: {
    flex: 1,
  },
  innerContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
    backgroundColor: '#FFF',
    paddingBottom: 20,
  },
  connectionsContainer: {
    backgroundColor: '#4AA9C5',
    padding: 10,
    borderRadius: 5,
    width: '100%'
  }
})
