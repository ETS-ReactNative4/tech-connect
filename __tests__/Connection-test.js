import React from 'react'
import { Connection, mapStateToProps } from '../src/Connection'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'


describe('Connection', () => {
  let wrapper
  let mockViewProfile
  let mockApiKey

  beforeEach(() => {
    mockViewProfile = jest.fn()
    mockApiKey - 123998347574354
    const mockConnection = {
      id: 3,
      name: 'Howard',
      job_title: 'Astronomer',
      city: 'Denver'
    }
    wrapper = shallow(<Connection connection={ mockConnection } viewProfile={ mockViewProfile } />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should call viewProfile on press of the name', () => {
    wrapper.find(TouchableOpacity).first().simulate('press')

    expect(mockViewProfile).toHaveBeenCalled()
  })

  it('mapStateToProps should return an object with an api_key', () => {
    const mockState = {
      isLoading: false,
      hasErrored: false,
      user: {
        api_key: 123456789
      }
    }
    const expected = {
      api_key: 123456789
    }

    const result = mapStateToProps(mockState)

    expect(result).toEqual(expected)
  })
})