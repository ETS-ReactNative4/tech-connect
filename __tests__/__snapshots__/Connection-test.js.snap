// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection should match the snapshot 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 5,
    }
  }
>
  <Component
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 40,
        "height": 80,
        "width": 80,
      }
    }
  />
  <Component
    style={
      Object {
        "marginLeft": 5,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Component
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Howard
      </Component>
    </TouchableOpacity>
    <Component
      style={
        Object {
          "color": "gray",
          "marginTop": 1,
        }
      }
    >
      Astronomer
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 3,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#4AA9C5"
        name="map-pin"
        size={18}
        style={
          Object {
            "marginRight": 7,
          }
        }
      />
      <Component
        style={
          Object {
            "color": "gray",
            "fontSize": 14,
            "marginTop": 2,
          }
        }
      >
        Denver
      </Component>
    </Component>
  </Component>
</Component>
`;
